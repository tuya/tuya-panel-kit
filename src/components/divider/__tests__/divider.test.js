import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Divider from '../index';

enzyme.configure({ adapter: new Adapter() });

describe('Divider Component', () => {
  it('basic render', () => {
    const component = shallow(<Divider />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('render with props', () => {
    const component = shallow(
      <Divider
        width={200}
        style={{
          flexDirection: 'column',
        }}
        height={2}
        color="#ff0000"
        visible={false}
      />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });
});
