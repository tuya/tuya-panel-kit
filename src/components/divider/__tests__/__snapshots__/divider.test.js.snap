// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider Component basic render 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#DBDBDB",
      },
      null,
      Object {
        "height": 0.5,
      },
    ]
  }
/>
`;

exports[`Divider Component render with props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#DBDBDB",
      },
      Object {
        "flexDirection": "column",
      },
      Object {
        "width": 0.5,
      },
      Object {
        "width": 200,
      },
      Object {
        "height": 2,
      },
      Object {
        "backgroundColor": "#ff0000",
      },
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
/>
`;
